framework/src/preconditioners/FiniteDifferencePreconditioner.C

```diff
@@ -83,12 +70,16 @@ FiniteDifferencePreconditioner::FiniteDifferencePreconditioner(const InputParame
 
     // off-diagonal entries
     std::vector<std::vector<unsigned int>> off_diag(n_vars);
-    for (unsigned int i = 0; i < getParam<std::vector<std::string>>("off_diag_row").size(); i++)
+    for (unsigned int i = 0;
+         i < getParam<std::vector<NonlinearVariableName>>("off_diag_row").size();
+         i++)
```

Comment:
```suggestion
    for (const auto i: index_range(getParam<std::vector<NonlinearVariableName>>("off_diag_row")))
```
would be better here as it guarantees single evaluation of the loop bounds... and also looks more readable